sdk/python/feast/permissions/auth/kubernetes_token_parser.py

```diff
@@ -41,10 +42,14 @@ async def user_details_from_access_token(self, access_token: str) -> User:
         current_user = f"{sa_namespace}:{sa_name}"
         logging.info(f"Received request from {sa_name} in {sa_namespace}")
 
-        roles = self.get_roles(sa_namespace, sa_name)
-        logging.info(f"SA roles are: {roles}")
+        intra_communication_base64 = os.getenv("INTRA_COMMUNICATION_BASE64")
+        if sa_name is not None and sa_name == intra_communication_base64:
+            return User(username=sa_name, roles=[])
```

Comment:
Wouldn't it be cleaner to introduce a magic role here (`admin` for example) and then security manager will check for that role instead of dealing with a username?

One upside there is that a user will also be able to take advantage of this and assign that `admin` role to normal users if that's the behavior they want. wdyt?

Reply:
You already raised the point in the past ([here](https://github.com/feast-dev/feast/issues/4198#issuecomment-2187122275)), but the reason for not taking this approach is to prevent users from assigning the same role to their own `Permission`s. This could be challenging both to explain and to implement.

Reply:
Fair enough, I just think that the ability to assign superadmin role to specific users might be a feature request from users later on. Still, no reason for us to implement it now.

Reply:
Agree. ATM the `superadmin` requires a manual definition like:
```py
Permission(
    name="admin_permission",
    types=ALL_RESOURCE_TYPES,
    policy=RoleBasedPolicy(roles=["superadmin"]),
    actions=ALL_ACTIONS
)
```
We should expose a simplified configuration for this user with extended privileges, maybe another field under the `auth` section of `feature_store.yaml`. As you suggested, let's wait for feedback from the early adopters 😉